optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/ListUnassignMove.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.heuristic.selector.move.generic.list;
+
+import java.util.Objects;
+
+import org.optaplanner.core.api.score.director.ScoreDirector;
+import org.optaplanner.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+import org.optaplanner.core.impl.heuristic.move.AbstractMove;
+import org.optaplanner.core.impl.score.director.InnerScoreDirector;
+
+public class ListUnassignMove<Solution_> extends AbstractMove<Solution_> {
+
+    private final ListVariableDescriptor<Solution_> variableDescriptor;
+    private final Object sourceEntity;
+    private final int sourceIndex;
+
+    public ListUnassignMove(
+            ListVariableDescriptor<Solution_> variableDescriptor,
+            Object sourceEntity, int sourceIndex) {
+        this.variableDescriptor = variableDescriptor;
+        this.sourceEntity = sourceEntity;
+        this.sourceIndex = sourceIndex;
+    }
+
+    @Override
+    protected AbstractMove<Solution_> createUndoMove(ScoreDirector<Solution_> scoreDirector) {
+        return null;
```

Comment:
We better implement this. For undo tabu search maybe.

Reply:
-1.

Assign and unassign moves are currently only used in CH, so they cannot be used in LS with undo move tabu.

I'll add a comment explaining that an undo move is never being undone.

> Undo move tabu makes the undo move of recent steps tabu.

So what we really need is to be able to say whether a `ListAssignMove` and a `ListUnassignMove` are equal when assign and unassign moves become used in LS.